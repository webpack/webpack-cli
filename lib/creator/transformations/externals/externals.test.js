"use strict";

const defineTest = require("../../../transformations/defineTest");

defineTest(__dirname, "externals", "externals-0", /react/);
defineTest(__dirname, "externals", "externals-1", {
	jquery: "'jQuery'",
	react: "'react'"
});

defineTest(__dirname, "externals", "externals-1", "myObj");

defineTest(__dirname, "externals", "externals-1", {
	jquery: "'jQuery'",
	react: "reactObj"
});

defineTest(__dirname, "externals", "externals-1", {
	jquery: "'jQuery'",
	react: ["reactObj", "path.join(__dirname, 'app')", "'jquery'"]
});

defineTest(__dirname, "externals", "externals-1", {
	lodash: {
		commonjs: "'lodash'",
		amd: "'lodash'",
		root: "'_'"
	}
});

defineTest(__dirname, "externals", "externals-1", {
	lodash: {
		commonjs: "lodash",
		amd: "hidash",
		root: "_"
	}
});

defineTest(__dirname, "externals", "externals-1", [
	{
		a: "false",
		b: "true",
		"'./ext'": "./hey"
	},
	"function(context, request, callback) {" +
		"if (/^yourregex$/.test(request)){" +
		"return callback(null, 'commonjs ' + request);" +
		"}" +
		"callback();" +
		"}"
]);

defineTest(__dirname, "externals", "externals-1", [
	"myObj",
	"function(context, request, callback) {" +
		"if (/^yourregex$/.test(request)){" +
		"return callback(null, 'commonjs ' + request);" +
		"}" +
		"callback();" +
		"}"
]);
