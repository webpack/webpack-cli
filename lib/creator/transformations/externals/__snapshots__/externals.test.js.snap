// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`externals transforms correctly using "externals-0" data 1`] = `
"module.exports = {
    entry: 'index.js',
    output: {
		filename: 'bundle.js'
	},

    externals: /react/
}
"
`;
